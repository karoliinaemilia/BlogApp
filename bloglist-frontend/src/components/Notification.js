//@flow

import React from 'react'
import { Message } from 'semantic-ui-react'

const Notification = ({ notification }) => {
  if (notification === null) {
    return null
  }

  if (notification.type === 'info') {
    return (
      <Message success>
        {notification.message}
      </Message>
    )
  } 

  return (
    <Message negative>
      {notification.message}
    </Message>
  )
}

export default Notification