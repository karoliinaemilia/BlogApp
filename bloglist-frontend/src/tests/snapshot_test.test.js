import React from 'react'
import Notification from '../components/Notification'
import Togglable from '../components/Togglable'
import renderer from 'react-test-renderer'


const notification = {
  type: 'info',
  message: 'hello'
}

it('renders correctly', () => {
  const tree = renderer
    .create(<Notification notification={notification}/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly', () => {
  const tree = renderer
    .create(<Togglable buttonLabel={'definitely a label'}/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})