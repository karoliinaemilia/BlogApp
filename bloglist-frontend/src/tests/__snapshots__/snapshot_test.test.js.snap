// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui success message"
>
  hello
</div>
`;

exports[`renders correctly 2`] = `
<div>
  <div
    style={
      Object {
        "display": "",
      }
    }
  >
    <button
      className="ui black button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      definitely a label
    </button>
  </div>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <br />
    <button
      className="ui button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      cancel
    </button>
  </div>
</div>
`;
